docs/tutorials/HPCIntro.md

```diff
@@ -132,6 +132,31 @@ backend.providers.SGE.config {
 }
 ```
 
+##### Call Caching based on runtime attributes
+
+The rules for call caching in HPC backends are:
+* `docker`: Will be considered when call caching.
+* Memory options: Will *not* be considered when call caching.
+* CPU options: Will *not* be considered when call caching.
+* Custom Attributes: Will *not* be considered when call caching (by default).
+ 
+Although custom attributes will not be considered when call caching by default, you can override this in a `runtime-attributes-for-caching` section. Eg:
+```hocon
+backend.providers.SGE.config {
+  runtime-attributes = """
+  String sge_queue = "short"
+  String singularity_image
+  # ...
+  """
+  runtime-attributes-for-caching {
+    sge_queue: false
+    singularity_image: true
+  }
+}
+```
+
+Note: Only *custom* attributes can be altered like this. Memory, CPU and docker will always have their default cache-consideration behavior.
```

Comment:
I'd also make a note that unlike `runtime { docker: ... }`, this PR _only_ checks that the `runtime { singularity: ... }` string is the same cached string.

The `docker` runtime attributes go through an extra effort, even on the local backend, dereferencing the image-name-string converting it to a hash. This happens using the on-machine docker daemon. This type of lookup will NOT happen for `singularity` image names, meaning if you tag a new image as "my-latest-version" you will still end up call-caching even though a new image available.

https://github.com/broadinstitute/cromwell/blob/51/dockerHashing/src/test/scala/cromwell/docker/local/DockerCliSpec.scala

One could build that extra-hop plugin interface... but I really think a warning is enough for now / iterative development.